toolkit-for-ynab/toolkit-for-ynab
# Safari web extension

**Explanation of Bugfix/Feature/Modification:**
Creates an Xcode project that converts the existing extension into a Safari Web Extension. This Xcode project uses whatever is in the `dist/extension` directory to generate the extension so the project will update itself whenever you call the build scripts to generate a new build of the extension.

To publish the extension, you will need an Apple Developer account, Xcode (to create the bundle) and it will need to be put on the Mac App Store.

**Prerequisites:**
- Mac OS 10.15 (and up)
- Safari 14
- Xcode 12

### To Test:
1. Run the build scripts to generate a build (any build is fine but you will of course want the production build when you go to publish the extension)
2. Open Safari
3. In Preferences > Advanced, turn on the Develop menu
4. In the Develop menu, click on Allow Unsigned Extensions
5. Run the Xcode project
6. A small window will pop up telling you the extension is on and that you need to turn it on in the safari extension preferences
7. Turn it on and you will see it appear next to your navigation bar. 

It will only activate when the YNAB webpage is active.

![Screen Shot 2020-10-16 at 4 15 53 PM](https://user-images.githubusercontent.com/9014774/96305204-83dcde80-0fcb-11eb-94df-7a3ddf9b66e0.png)
